### PR TITLE
Different tstarts

### PR DESCRIPTION
**Pull request by huzafahyde**
Added capability for different tstarts. Works with negative and positive values

Addresses #11

### CODE REVIEW

**Comment by proloyd** [comment]
@huzafahyde please all add couple of tests to check if the implementation is working for both positive and negative values.

**Comment by huzafahyde** [comment]
@proloyd fixed issue with slight misestimation, and added tests for a model with multiple tstarts (both positive and negative)

**Comment by proloyd** [comment]
@huzafahyde looks good. Could you please also update the docstring to reflect the changes.
https://github.com/proloyd/neuro-currentRF/blob/1295905d04851f4b84caad8c2f0b4b4949c3fe9f/ncrf/_ncrf.py#L41

https://github.com/proloyd/neuro-currentRF/blob/1295905d04851f4b84caad8c2f0b4b4949c3fe9f/ncrf/_model.py#L223

File: ncrf/_model.py [hunk 1]
```diff
@@ -287,10 +285,13 @@ def add_data(self, meg, stim):
                 stim_dims.append(dim)
             else:
                 raise ValueError(f"stim={stim}: stimulus with more than 2 dimensions")
-
+
```
Comment (proloyd) [inline ncrf/_model.py:302]: Ordering could be: `tstart`, followed by `tstop`. We are following that order in the rest of the file.

**Review by proloyd — CHANGES_REQUESTED** [review]

**Comment by huzafahyde** [comment]
@proloyd Fixed the -ve tstart estimation and added tests

File: ncrf/_model.py [hunk 2]
```diff
@@ -124,15 +125,33 @@ def covariate_from_stim(stims, Ms):
         ws.append(w)
     w = ws[0] if len(ws) == 1 else np.concatenate(ws, 0)
 
+    # adds tstarts into array for multi dimensional stimuli
+    if len(start) != len(w):
+        ndim_starts = np.full(stim.ndim - 1, start[i])
+        start = np.insert(start, i, ndim_starts)
+
+    assert len(w) == len(Ms) == len(start), f"Length of w ({len(w)}), Ms ({len(Ms)}), and start ({len(start)}) should be equal"
+
```
Comment (christianbrodbeck) [inline ncrf/_model.py]: Shouldn't this happen inside the `for` loop starting on line 119?

File: ncrf/_model.py [hunk 3]
```diff
@@ -124,15 +125,33 @@ def covariate_from_stim(stims, Ms):
...
     length = w.shape[1]
     Y = []
     M_ = max(Ms)
     for j, M in zip(range(w.shape[0]), Ms):
```
Comment (christianbrodbeck) [inline ncrf/_model.py]:
```suggestion
    for j, M in enumerate(Ms):
```

**Review by christianbrodbeck — CHANGES_REQUESTED** [review]
I added two comments, looks good otherwise! I'll leave it to @proloyd to have another look.

File: ncrf/_model.py (hunk 2 above)
Comment (proloyd) [reply ncrf/_model.py]: I have the same comment.

File: ncrf/_model.py [hunk 4]
```diff
@@ -99,23 +98,25 @@ def proxg_group_opt(z, mu):
     return z
 
 
-def covariate_from_stim(stims, Ms):
+def covariate_from_stim(stims, Ms, start):
```
Comment (proloyd) [inline ncrf/_model.py:101]: @huzafahyde could you add a quick standalone test i.e. `test_covariate_from_stim()`? possible test cases:
- list of stims vs 2D stim (i.e., result of `eelbrain.concatenate()` on list of stims) vs explicit looping over stims
- start 0 vs. start -50. The covariate in the former case should be recoverable from the covariate of the later case if indexed properly.
Thanks!

**Review by proloyd — CHANGES_REQUESTED** [review]

File: ncrf/_model.py (hunk 4 above)
Comment (proloyd) [reply ncrf/_model.py:101]: Thanks @huzafahyde

**Review by proloyd — APPROVED** [review]
Looks good.

**Comment by proloyd** [comment]
Thanks @huzafahyde !
